src/box/lua/schema.lua

```diff
@@ -1587,21 +1571,19 @@ box.schema.index.create = function(space_id, name, options)
     if index_opts.func ~= nil and type(index_opts.func) == 'string' then
         index_opts.func = func_id_by_name(index_opts.func)
     end
-    local sequence_proxy = space_sequence_alter_prepare(format, parts, options,
-                                                        space_id, iid,
-                                                        space.name, name)
     _index:insert{space_id, iid, name, options.type, index_opts, parts}
-    space_sequence_alter_commit(sequence_proxy)
+    space_sequence_alter(format, parts, options, space_id,
+                         iid, space.name, name)
```

Comment:
Let's do all the sequence-related checks *before* altering the index definition. Rationale: index rebuild may take quite a lot of time; it'd be nice to fail early if sequence parameters are invalid.

Reply:
That was a serious overlook, fixed.